src/connection.cc

```diff
@@ -666,68 +553,59 @@ NAN_METHOD(Connection::Notifies) {
 
   if(msg == NULL) {
     LOG("No notification");
-    NanReturnUndefined();
+    return;
   }
 
-  v8::Local<v8::Object> result = NanNew<v8::Object>();
-  result->Set(NanNew("relname"), NanNew(msg->relname));
-  result->Set(NanNew("extra"), NanNew(msg->extra));
-  result->Set(NanNew("be_pid"), NanNew(msg->be_pid));
+  v8::Local<v8::Object> result = Nan::New<v8::Object>();
+  Nan::Set(result, Nan::New("relname").ToLocalChecked(), Nan::New(msg->relname).ToLocalChecked());
+  Nan::Set(result, Nan::New("extra").ToLocalChecked(), Nan::New(msg->extra).ToLocalChecked());
+  Nan::Set(result, Nan::New("be_pid").ToLocalChecked(), Nan::New(msg->be_pid));
 
   PQfreemem(msg);
 
-  NanReturnValue(result);
+  info.GetReturnValue().Set(result);
 };
 
 NAN_METHOD(Connection::PutCopyData) {
-  NanScope();
   LOG("Connection::PutCopyData");
 
   Connection* self = THIS();
 
-  v8::Handle<v8::Object> buffer = args[0].As<v8::Object>();
+  v8::Local<v8::Object> buffer = info[0].As<v8::Object>();
 
   char* data = node::Buffer::Data(buffer);
   int length = node::Buffer::Length(buffer);
 
   int result = PQputCopyData(self->pq, data, length);
 
-  NanReturnValue(NanNew<v8::Number>(result));
+  info.GetReturnValue().Set(result);
 }
 
 NAN_METHOD(Connection::PutCopyEnd) {
-  NanScope();
   LOG("Connection::PutCopyEnd");
 
   Connection* self = THIS();
 
   //optional error message
 
-  bool sendErrorMessage = args.Length() > 0;
-  char* msg = NULL;
+  bool sendErrorMessage = info.Length() > 0;
+  Nan::Utf8String msg(info[0]);
   if(sendErrorMessage) {
-    msg = NewCString(args[0]);
-    TRACEF("Connection::PutCopyEnd:%s\n", msg);
+    TRACEF("Connection::PutCopyEnd:%s\n", *msg);
   }
 
-  int result = PQputCopyEnd(self->pq, msg);
+  int result = PQputCopyEnd(self->pq, *msg);
```

Comment:
In the non-cancel case msg is still not null, that's why the copy-in tests fail.
Passing in a char msg = NULL makes them succeed.